core/src/main/java/net/adoptopenjdk/icedteaweb/client/controlpanel/panels/ServerWhitelistPanel.java

```diff
@@ -46,9 +56,97 @@ public ServerWhitelistPanel(final DeploymentConfiguration config) {
 
         Assert.requireNonNull(config, "config");
 
-        final List<String> whitelist = config.getPropertyAsList(KEY_SECURITY_SERVER_WHITELIST);
-        final JList<String> jList = new JList<>(new Vector<>(whitelist));
-        jList.setFixedCellHeight(20);
-        add(new JScrollPane(jList), BorderLayout.CENTER);
+        final List<WhitelistEntry> whitelist = UrlWhiteListUtils.getApplicationUrlWhiteList();
+
+        final JTable table = new JTable(createTableModel(whitelist));
+        table.getTableHeader().setReorderingAllowed(false);
+        table.setFillsViewportHeight(true);
+        table.setAutoResizeMode(JTable.AUTO_RESIZE_ALL_COLUMNS);
+        table.setAutoCreateRowSorter(true);
+
+        final TableColumnModel colModel = table.getColumnModel();
+        colModel.getColumn(0).setPreferredWidth(100);
+        colModel.getColumn(1).setPreferredWidth(250);
+
+        final ImageIcon icon = SunMiscLauncher.getSecureImageIcon("net/sourceforge/jnlp/resources/warn16.png");
+        final DefaultTableCellRenderer cellRenderer = new DefaultTableCellRenderer() {
+            @Override
+            public Component getTableCellRendererComponent(JTable table, Object value, boolean isSelected, boolean hasFocus, int row, int column) {
+                final WhitelistEntryState wleState = (WhitelistEntryState) value;
+                if (wleState != null) {
+                    setText(wleState.getMessage());
+                    if (!wleState.isValid()) {
+                        setIcon(icon);
+                    } else {
+                        setIcon(null);
+                    }
+                } else {
+                    setText(null);
+                    setIcon(null);
+                }
+                if (isSelected) {
+                    setBackground(table.getSelectionBackground());
```

Comment:
is this background / foreground setting really needed? I assume the default renderer already does this